content/blog/2023/2023-05-18-operators-in-typescript.md

```diff
@@ -0,0 +1,177 @@
+
+~~~ts
+let x = 10;
+let y = 5;
+
+console.log(x == y); // false
+console.log(x === "10"); // false (different data types)
+console.log(x != y); // true
+console.log(x !== "10"); // true (different data types)
+console.log(x > y); // true
+console.log(x < y); // false
+console.log(x >= y); // true
+console.log(x <= y); // false
+~~~
+
+**Equality Operator (`==`):** This operator compares two values but doesn't consider their data types. If the values are equal, it returns true. Otherwise, it returns false.
+
+**Strict Equality Operator (`===`):** This operator compares two values for equality, and it considers their data types. If the values are equal in value and type, it returns true. Otherwise, it returns false.
+
+**Inequality Operator (`!=`):** This operator compares two values for inequality. If the values are not equal, it returns true. Otherwise, it returns false.
+
+**Strict Inequality Operator (`!==`):** This operator compares two values for inequality, and it considers their data types. If the values are not equal in value or type, it returns true. Otherwise, it returns false.
+
+**Greater Than Operator (`>`):** This operator checks if the left operand is greater than the right operand. If it is true, it returns true. Otherwise, it returns false.
+
+**Less Than Operator (`<`):** This operator checks if the left operand is less than the right operand. If it is true, it returns true. Otherwise, it returns false.
+
+**Greater Than or Equal To Operator (`>=`):** This operator checks if the left operand is greater than or equal to the right operand. If it is true, it returns true. Otherwise, it returns false.
+
+**Less Than or Equal To Operator (`<=`):** This operator checks if the left operand is less than or equal to the right operand. If it is true, it returns true. Otherwise, it returns false.
+
+### Difference Between Equality and Strict Equality Operator:
+When it comes to comparing values, it is essential to understand the difference between the equality (`==`) and strict equality (`===`) operators. The equality operator only compares the values of the operands, while the strict equality operator compares both the values and types of the operands. Let’s take a look at this code:
+
+~~~ts
+let x = 10;
+let y = 5;
+console.log(x == y); // false
+console.log(x == 10); // true
+console.log(x === y); // false (different value)
+console.log(x === "10"); // false (different data types)
+console.log(x === 10); // true (same data types)
+~~~
+
+In the given example, the expression `x == y` returns false because `x` and `y` are not equal. The expression `x == 10` returns true because `x` is equal to `10`. The expression `x === y` returns false because `x` and `y` are of different values. The expression `x === '10'` returns false because `x` is a number and `'10'` is a string. Finally, the expression `x === 10` returns true because both `x` and `10` are of the same data type (number) and have the same value.
```

Comment:
> because `x` and `y` are not equal

because the **values** of `x` and `y` are not equal! Right? Otherwise, you're just repeating what `x == y` says

Reply:
Maybe put each sentence in this paragraph in a bullet point in a list?

Reply:
> The expression `x === y` returns false because `x` and `y` are of different values
Would be clearer if you said " ... of the same type, but of different values".

Reply:
> The expression x === '10' returns false because x is a number and '10' is a string
Would be clearer if you included "...is a string (i.e. they are of different types)"